guides/keras_cv/cut_mix_mix_up_and_rand_augment.py

```diff
@@ -0,0 +1,398 @@
+
+
+visualize_dataset(train_dataset, title="Before Augmentation")
+
+"""
+Great!  Now we can move onto the augmentation step.
+"""
+
+"""
+## RandAugment
+"""
+
+"""
+RandAugment performs a standard set of perturbations to an image based on the parameters
+`magnitude`, `magnitude_stddev`, `augmentations_per_image` and `rate`.  Each of these
+parameters should be tuned to fit your specific case.  You can read more about these
+parameters in the [RandAugment API documentation](/api/keras_cv/layers/rand_augment).
+
+To remedy this issue KerasCV offers RandAugment, a general purpose image data
+augmentation algorithm.  RandAugment has been shown to provide improved image
+classification results across numerous datasets, and only has two hyperparameters to
+configure.
+
+To use `RandAugment` you need to provide a few values:
+
+- `value_range` describes the range of values covered in your images
+- `magnitude` is a value between 0 and 10, describing the strength of the perturbations
+applied
+- `num_layers` is an integer telling the layer how many augmentations to apply to each
+individual image
+- (Optional) `magnitude_standard_deviation` allows `magnitude` to be randomly sampled
+from a distribution with a standard deviation of `magnitude_standard_deviation`
+- (Optional) `probability_to_apply` indicates the probability to apply the augmentation
+applied at each layer.
+
+Let's use KerasCV's RandAugment implementation.
+"""
+
+rand_augment = keras_cv.layers.RandAugment(
+    value_range=(0, 255),
+    augmentations_per_image=3,
+    magnitude=0.3,
+    magnitude_stddev=0.2,
+    rate=0.5,
+)
+
+
+def apply_rand_augment(inputs):
+    inputs["images"] = rand_augment(inputs["images"])
+    return inputs
+
+
+train_dataset = load_dataset().map(
+    apply_rand_augment, num_parallel_calls=tf.data.AUTOTUNE
+)
+
+"""
+And finally lets inspect some of the results:
+"""
+
+visualize_dataset(train_dataset, title="After RandAugment")
+
+"""
+Try tweaking the magnitude settings to see a wider variety of results.
+"""
+
+"""
+## CutMix and MixUp: Generate High Quality Inter-class Examples
+
+
+`CutMix` and `MixUp` allow us to produce inter-class examples.  CutMix randomly cuts out
+portions of one image and places them over another, and MixUp interpolates the pixel
+values between two images.  Both of these prevent the model from overfitting the
+training distribution and improve the likelihood that the model can generalize to out of
+distribution examples.  Additionally, CutMix prevents your model from over-relying on
+any particular feature to perform its classifications.
+
+In this example, we will use `CutMix` and `MixUp` independently in a manually created
+preprocessing pipeline.  In most state of the art pipelines images are randomly
+augmented by either `CutMix`, `MixUp`, or neither.  The function below implements this
+in an equal 1/3 split.
+
+Note that our `cut_mix_and_mix_up` function is annotated with a `tf.function` to ensure
+optimal performance.
```

Comment:
Very nice!

A couple of thoughts:

* Links to the original works. 
* Links to the keras.io examples of CutMix and MixUp (https://keras.io/examples/vision/cutmix, https://keras.io/examples/vision/mixup) (optional)

Reply:
Let's wait to link to these until we migrate these guides to use KerasCV.

Reply:
Linked to original papers though.